crates/ruff/src/rules/ruff/rules/explicit_f_string_type_conversion.rs

```diff
@@ -41,72 +46,113 @@ impl AlwaysAutofixableViolation for ExplicitFStringTypeConversion {
     }
 }
 
-/// RUF010
-pub(crate) fn explicit_f_string_type_conversion(
-    checker: &mut Checker,
+fn fix_explicit_f_string_type_conversion(
     expr: &Expr,
-    formatted_value: &Expr,
-    conversion: ast::Int,
-) {
-    // Skip if there's already a conversion flag.
-    if conversion != ast::ConversionFlag::None as u32 {
-        return;
+    formatted_values: &[usize],
+    locator: &Locator,
+    stylist: &Stylist,
+) -> Result<Fix> {
+    // Replace the call node with its argument and a conversion flag.
+    let range = expr.range();
+    let content = locator.slice(range);
+    let mut expression = match_expression(content)?;
+    let formatted_string = match_formatted_string(&mut expression)?;
+
+    for index in formatted_values {
+        let mut formatted_string_expression =
+            match_formatted_string_expression(&mut formatted_string.parts[*index])?;
+        let call = match_call(&mut formatted_string_expression.expression)?;
+        let name = match_name(&mut call.func)?;
+        match name.value {
+            "str" => {
+                formatted_string_expression.conversion = Some("s");
+            }
+            "repr" => {
+                formatted_string_expression.conversion = Some("r");
+            }
+            "ascii" => {
+                formatted_string_expression.conversion = Some("a");
+            }
+            _ => unreachable!(),
+        }
+        formatted_string_expression.expression = call.args[0].value.clone();
     }
 
-    let ExprKind::Call(ast::ExprCall {
-        func,
-        args,
-        keywords,
-    }) = &formatted_value.node else {
-        return;
+    let mut state = CodegenState {
+        default_newline: &stylist.line_ending(),
+        default_indent: stylist.indentation(),
+        ..CodegenState::default()
     };
+    expression.codegen(&mut state);
 
-    // Can't be a conversion otherwise.
-    if args.len() != 1 || !keywords.is_empty() {
-        return;
-    }
-
-    let ExprKind::Name(ast::ExprName { id, .. }) = &func.node else {
-        return;
-    };
+    Ok(Fix::automatic(Edit::range_replacement(
+        state.to_string(),
+        range,
+    )))
```

Comment:
(An alternative fix would be to just use `"foo"` directly given `str("foo")`, and then add the `!r` manually, instead of using `unparse`.) What you have here looks more robust, but the direct fix would also be safe (and is faster since we don't have to reparse the f-string). I leave it up to you!

Reply:
I just did that.